state/block.go

```diff
@@ -194,13 +195,30 @@ func (bs *blockState) SetBabeHeader(epoch uint64, slot uint64, blockData babe.Ba
 	return err
 }
 
-// AddBlock will add a block into the DB
-func (bs *blockState) AddBlock(block types.BlockHeader) error {
+// AddBlock will set the latestBlock in blockState DB
+func (bs *blockState) AddBlock(block types.Block) error {
+	blockHeader := *block.Header
+
 	// Set the latest block
-	if block.Number.Cmp(bs.latestBlock.Number) == 1 {
-		bs.latestBlock = block
+	if reflect.DeepEqual(bs.latestBlock, types.BlockHeader{}) {
+		bs.latestBlock = blockHeader
+	} else if blockHeader.Number != nil && blockHeader.Number.Cmp(bs.latestBlock.Number) == 1 { // If the new block has a number greater than current latestBlock
```

Comment:
if blockHeader.Number == nil we probably want to return an error, since that isn't a valid block

Reply:
It will be cleaner, but returning a pointer may cause side effects later on when we modify the header for consensus changes.

Reply:
I'll make some changes here, your comment also made me see the latestBlock is actually not a block, but a header, and its confusing so I will update it a bit.